picard/ui/infodialog.py

```diff
@@ -99,18 +99,37 @@ class InfoDialog(PicardDialog):
     def __init__(self, obj, parent=None):
         PicardDialog.__init__(self, parent)
         self.obj = obj
+        self.images = []
+        self.existing_images = []
         self.ui = Ui_InfoDialog()
         self.display_existing_artwork = False
-        if (isinstance(obj, File)
-            and isinstance(obj.parent, Track)
-            or isinstance(obj, Track)):
+
+        def get_image_type(image):
+            return image.types_as_string()
+
+        if (isinstance(obj, File) and
+                isinstance(obj.parent, Track) or
+                isinstance(obj, Track)):
             # Display existing artwork only if selected object is track object
             # or linked to a track object
-            if getattr(obj, 'orig_metadata', None) is not None:
+            if (getattr(obj, 'orig_metadata', None) is not None and
```

Comment:
Indeed, I tried several options there until reaching that point where it can be easily merged